lib/posix/mqueue.c

```diff
@@ -341,6 +347,60 @@ int mq_setattr(mqd_t mqdes, const struct mq_attr *mqstat,
 	return 0;
 }
 
+/**
+ * @brief Notify process that a message is available.
+ *
+ * See IEEE 1003.1
+ */
+int mq_notify(mqd_t mqdes, const struct sigevent *notification)
+{
+	mqueue_desc *mqd = (mqueue_desc *)mqdes;
+
+	if (mqd == NULL) {
+		errno = EBADF;
+		return -1;
+	}
+
+	mqueue_object *msg_queue = mqd->mqueue;
+
+	if (notification == NULL) {
+		if ((msg_queue->not.sigev_notify & SIGEV_MASK) == 0) {
+			errno = EINVAL;
+			return -1;
+		}
+		remove_notification(msg_queue);
+	} else {
+		if ((msg_queue->not.sigev_notify & SIGEV_MASK) != 0) {
+			errno = EBUSY;
+			return -1;
+		}
+		if (notification->sigev_notify == SIGEV_SIGNAL) {
+			errno = ENOSYS;
+			return -1;
+		}
+		k_sem_take(&mq_sem, K_FOREVER);
+		memcpy(&msg_queue->not, notification, sizeof(struct sigevent));
+		k_sem_give(&mq_sem);
+	}
+
+	return 0;
+}
+
+static void *mq_notify_thread(void *arg)
+{
+	mqueue_object *mqueue = (mqueue_object *)arg;
+	struct sigevent *sevp = &mqueue->not;
+
+	pthread_detach(pthread_self());
```

Comment:
Aside from calling `pthread_detach()`, I think it might be necessary to set the cancel behaviour to asynchronous. Otherwise, calling `pthread_cancel()` might have no (immediate) effect.

https://github.com/zephyrproject-rtos/zephyr/pull/67871 for reference, since there was a similar recent bug in `timer.c`